packages/vite-plugin-svelte/src/index.js

```diff
@@ -67,14 +63,31 @@ export function svelte(inlineOptions) {
 				log.debug('additional vite config', extraViteConfig, 'config');
 				return extraViteConfig;
 			},
-			// @ts-ignore Allow exist in vite 6
-			configEnvironment(name, config) {
+
+			// @ts-ignore This hook only works in Vite 6
+			async configEnvironment(name, config, opts) {
 				config.resolve ??= {};
+
+				// Emulate Vite default fallback for `resolve.mainFields` if not set
+				if (config.resolve.mainFields == null) {
+					// These exports only exist in Vite 6
+					const { defaultClientMainFields, defaultServerMainFields } = await import('vite');
+					if (name === 'client' || opts.isSsrTargetWebworker) {
+						config.resolve.mainFields = [...defaultClientMainFields];
+					} else {
+						config.resolve.mainFields = [...defaultServerMainFields];
+					}
+				}
+				config.resolve.mainFields.unshift('svelte');
+
+				// Emulate Vite default fallback for `resolve.conditions` if not set
 				if (config.resolve.conditions == null) {
-					if (name === 'client') {
-						config.resolve.conditions = [...VITE_CLIENT_RESOLVE_CONDITIONS];
+					// These exports only exist in Vite 6
+					const { defaultClientConditions, defaultServerConditions } = await import('vite');
+					if (name === 'client' || opts.isSsrTargetWebworker) {
```

Comment:
is name === 'client' a good way to check this? that would assume there can only be one client environment in an app.

Reply:
what if there was a tauri app and a web app at the same time, or two different clients?

Reply:
Ah true, I think it should check the `consumer` here, which is the same as Vite internally.

Reply:
I'm not sure if it's needed imo

Reply:
well how are others going to use these exposed constants then? you'll always want the ones applicable to the environment you are in, no?

Reply:
I'm not really sure what you mean. Do you have a pseudocode in mind for this?

Reply:
Looks like I need a combination of `config.consumer === 'client'` and `name === 'client'` checks to work, since `config.consumer` could be undefined but it's documented to fallback to `consumer = 'client'` if `name` is `'client'`